compose/config/config.py

```diff
@@ -133,6 +144,24 @@ def find(base_dir, filenames):
         [ConfigFile.from_filename(f) for f in filenames])
 
 
+def get_config_version(config_details):
+    def get_version(config):
+        validate_top_level_object(config)
```

Comment:
This `validate_top_level_object()` is now called twice for the same config. Once here, and again below as part of `process_config_file()`

It would be nice to only have it happen once, and I think this will actually fail now that `validate_top_level_object()` also validates that the top level keys are services names with values of objects.

I guess `validate_top_level_object()` could be split up again. The first step of `load()` could be to validate that every `ConfigFile` has a valid `config` dict. Then this `get_config_version()` can proceed without doing any more validation, and `process_config_file()` can call the rest of validation.